api/v1alpha1/author_webhook.go

```diff
@@ -0,0 +1,197 @@
+		}
+		author := releasePlanNew.GetLabels()[metadata.AuthorLabel]
+		// Attribution True -> True
+		if releasePlanOld.GetLabels()[metadata.AttributionLabel] == "true" {
+			// If changing author label to something beside current user, deny request
+			if author != "" && author != authorLabelValue(req.UserInfo.Username) &&
+				author != releasePlanOld.GetLabels()[metadata.AuthorLabel] {
+				return admission.Errored(http.StatusInternalServerError, errors.New("[labels] cannot change the author label if not setting"+
+					" it to the current user. Expected author is: "+authorLabelValue(req.UserInfo.Username)))
+			}
+			return admission.Allowed("Success")
+		}
+		if releasePlanNew.GetLabels()[metadata.AttributionLabel] == "true" &&
+			author != "" && author != authorLabelValue(req.UserInfo.Username) {
+			// Attribution False/Missing -> True, has request for different author, deny request
+			return admission.Errored(http.StatusInternalServerError, errors.New("[labels] cannot set attribution to true if not setting"+
+				" current user as author. Expected author is: "+authorLabelValue(req.UserInfo.Username)))
+		}
+		// Attribution False/Missing -> True, set author
+		return a.patchAuthorResponse(req.UserInfo.Username, &releasePlanNew)
+	default:
+		return admission.Allowed("Success")
+	}
+}
+
+// patchAuthorResponse returns an admission.Response that modifies the CR author label to be
+// the user provided.
+func (a *authorWebhook) patchAuthorResponse(username string, obj client.Object) admission.Response {
+	oldObj, err := json.Marshal(obj)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, errors.Wrap(err, "error encoding response object"))
+	}
+
+	objectLabels := make(map[string]string)
+	if obj.GetLabels() != nil {
+		objectLabels = obj.GetLabels()
+	}
+	objectLabels[metadata.AuthorLabel] = authorLabelValue(username)
+	obj.SetLabels(objectLabels)
+
+	newObj, err := json.Marshal(obj)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, errors.Wrap(err, "error encoding response object with author added"))
+	}
+
+	return admission.PatchResponseFromRaw(oldObj, newObj)
+}
+
+// removeAuthorResponse returns an admission.Response that removes the author label from the CR.
+func (a *authorWebhook) removeAuthorResponse(obj client.Object) admission.Response {
+	oldObj, err := json.Marshal(obj)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, errors.Wrap(err, "error encoding response object"))
+	}
+
+	delete(obj.GetLabels(), metadata.AuthorLabel)
+
+	newObj, err := json.Marshal(obj)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, errors.Wrap(err, "error encoding response object with author removed"))
+	}
+
+	return admission.PatchResponseFromRaw(oldObj, newObj)
+}
+
+// authorLabelValue takes a username and returns it in a form appropriate to use as a label value.
+func authorLabelValue(username string) string {
```

Comment:
Add a receiver here as well.

Reply:
It also needs a more descriptive name. Just call it `parseLabelValue` or whatever... Something that describes what it does.